code/game/objects/items/toys.dm

```diff
@@ -229,10 +229,64 @@
 	/// Sets to TRUE once the character using it hits something and realises it's not a real energy sword
 	var/pranked = FALSE
 
+/obj/item/toy/sword/attack(mob/target, mob/living/user)
+	if(!cigarette_lighter_act(user, target))
+		return ..()
+
+/obj/item/toy/sword/cigarette_lighter_act(mob/living/user, mob/living/target, obj/item/direct_attackby_item)
+	var/obj/item/clothing/mask/cigarette/cig = ..()
+	if(!cig)
+		return !isnull(cig)
+
+	if(!active)
+		to_chat(user, "<span class='warning'>You must activate [src] before you can light [cig] with it!</span>")
+		return TRUE
+
+	user.do_attack_animation(target)
+
+	// 1% chance to light the cig. Somehow...
+	if(prob(1))
+		if(target == user)
+			user.visible_message(
+				"<span class='warning'>[user] makes a violent slashing motion, barely missing [user.p_their()] nose as light flashes! \
+				[user.p_they(TRUE)] light[user.p_s()] [user.p_their()] [cig] with [src] in the process. Somehow...</span>",
+				"<span class='notice'>You casually slash [src] at [cig], lighting it with the blade. Somehow...</span>",
+				"<span class='danger'>You hear an energy blade slashing something!</span>"
+			)
+		else
+			user.visible_message(
+				"<span class='danger'>[user] makes a violent slashing motion, barely missing the nose of [target] as light flashes! \
+				[user.p_they(TRUE)] light[user.p_s()] [cig] in the mouth of [target] with [src] in the process. Somehow...</span>",
+				"<span class='notice'>You casually slash [src] at [cig] in the mouth of [target], lighting it with the blade. Somehow...</span>",
+				"<span class='danger'>You hear an energy blade slashing something!</span>"
+			)
+		playsound(user.loc, 'sound/weapons/blade1.ogg', 50, TRUE)
+		cig.light(user, target)
+		return TRUE
+
+	// Else, bat it out of the target's mouth.
+	if(target == user)
+		user.visible_message(
+			"<span class='warning'>[user] makes a violent slashing motion, barely missing [user.p_their()] nose as light flashes! \
+			[user.p_they(TRUE)] instead hit [cig], knocking it out of [user.p_their()] mouth and dropping it to the floor. What an idiot!</span>",
+			"<span class='warning'>You casually slash [src] at [cig], swatting it out of your mouth. Because it's not a real energy sword, dumbass!</span>",
```

Comment:
Things are looking good so far. This is more personal than anything, but I am not sure I like the `dumbass` and `idiot` portion of these visible messages. It should really just be scoped down to what is happening, and to let the players gauge and respond to the action instead. Also just not a big fan of calling the player a `dumbass` or an `idiot` like that in general.

Reply:
Cut that portion of the messages away.